File: solutions/fsharp/bob/1/Bob.fs
module Bob

open System

let isUpper (str: string) =
    str |> Seq.exists Char.IsLetter &&
    str |> Seq.filter Char.IsLetter |> Seq.forall Char.IsUpper

let response (input: string): string =
    let adjustedInput = input.Trim()
    if String.IsNullOrWhiteSpace(input) then
        "Fine. Be that way!"
    elif isUpper(input) && Seq.last adjustedInput = '?' then
        "Calm down, I know what I'm doing!"
    elif isUpper(input) then
        "Whoa, chill out!"
    elif Seq.last adjustedInput = '?' then
        "Sure."
    else
        "Whatever."

File: solutions/typescript/bob/1/bob.ts
export function hey(message: string): string {
  switch(true){
    case message.trim().endsWith("?") && message === message.toUpperCase() && /\p{L}/u.test(message):
      return "Calm down, I know what I'm doing!";
      break;
    case message.trim().endsWith("?"):
      return "Sure.";
      break;
    case message === message.toUpperCase() && /\p{L}/u.test(message):
      return "Whoa, chill out!";
      break;
    case message.trim() === "":
      return "Fine. Be that way!";
      break;
    default:
      return "Whatever.";
  }
}